state-chain/pallets/cf-broadcast/src/migrations/v3.rs

```diff
@@ -0,0 +1,271 @@
+	pub struct BroadcastAttempt<T: Config<I>, I: 'static> {
+		pub broadcast_attempt_id: BroadcastAttemptId,
+		pub transaction_payload: TransactionFor<T, I>,
+		pub threshold_signature_payload: PayloadFor<T, I>,
+		pub transaction_out_id: TransactionOutIdFor<T, I>,
+	}
+
+	#[derive(Clone, RuntimeDebug, PartialEq, Eq, Encode, Decode, TypeInfo)]
+	pub struct TransactionSigningAttempt<T: Config<I>, I: 'static> {
+		pub broadcast_attempt: BroadcastAttempt<T, I>,
+		pub nominee: T::ValidatorId,
+	}
+
+	#[frame_support::storage_alias]
+	pub type AwaitingBroadcast<T: Config<I>, I: 'static> =
+		StorageMap<Pallet<T, I>, Twox64Concat, BroadcastAttemptId, TransactionSigningAttempt<T, I>>;
+
+	#[frame_support::storage_alias]
+	pub type FailedBroadcasters<T: Config<I>, I: 'static> =
+		StorageMap<Pallet<T, I>, Twox64Concat, BroadcastId, Vec<<T as Chainflip>::ValidatorId>>;
+
+	#[frame_support::storage_alias]
+	pub type BroadcastRetryQueue<T: Config<I>, I: 'static> =
+		StorageValue<Pallet<T, I>, Vec<BroadcastAttempt<T, I>>>;
+}
+
+#[derive(Clone, Encode, Decode)]
+pub struct MigrationVerification {
+	broadcasts_with_failed_broadcasters: Vec<BroadcastId>,
+	broadcast_retry_queue: Vec<BroadcastId>,
+}
+pub struct Migration<T: Config<I>, I: 'static>(PhantomData<(T, I)>);
+
+impl<T: Config<I>, I: 'static> OnRuntimeUpgrade for Migration<T, I> {
+	fn on_runtime_upgrade() -> frame_support::weights::Weight {
+		// Awaiting broadcast: take the broadcast attempt with the highest attempt.
+		let mut latest_attempt = BTreeMap::new();
+		old::AwaitingBroadcast::<T, I>::iter_keys().for_each(
+			|old::BroadcastAttemptId { broadcast_id, attempt_count }| match latest_attempt
+				.get(&broadcast_id)
+			{
+				Some(attempt) =>
+					if attempt_count > *attempt {
+						latest_attempt.insert(broadcast_id, attempt_count);
+					},
+				None => {
+					latest_attempt.insert(broadcast_id, attempt_count);
+				},
+			},
+		);
+
+		// Migrate data from old storage into new storage.
+		latest_attempt.into_iter().for_each(|(broadcast_id, attempt_count)| {
+			if let Some(attempt) = old::AwaitingBroadcast::<T, I>::get(old::BroadcastAttemptId {
+				broadcast_id,
+				attempt_count,
+			}) {
+				AwaitingBroadcast::<T, I>::insert(
+					broadcast_id,
+					TransactionSigningAttempt {
+						broadcast_attempt: BroadcastAttempt {
+							broadcast_id,
+							transaction_payload: attempt.broadcast_attempt.transaction_payload,
+							threshold_signature_payload: attempt
+								.broadcast_attempt
+								.threshold_signature_payload,
+							transaction_out_id: attempt.broadcast_attempt.transaction_out_id,
+						},
+						nominee: attempt.nominee,
+					},
+				);
+			}
+		});
+
+		// Failed broadcaster storage: Vec<ValidatorId> -> BTreeSet<ValidatorId>
+		old::FailedBroadcasters::<T, I>::drain().for_each(|(broadcast_id, failed_broadcasters)| {
+			FailedBroadcasters::<T, I>::insert(
+				broadcast_id,
+				BTreeSet::from_iter(failed_broadcasters),
+			)
+		});
```

Comment:
```suggestion
		FailedBroadcasters::<T, I>::translate(|_, failed_broadcasters: Vec<T::ValidatorId>| {
			Some(BTreeSet::from_iter(failed_broadcasters))
		});
```
we only really need the `old` version in this case for the tests